source/dub/internal/git.d

```diff
@@ -102,6 +91,49 @@ private string determineVersionWithGitTool(NativePath path)
 	return null;
 }
 
+private string determineVersionFromGitDescribe(string describeOutput)
+{
+	import dub.semver : isValidVersion;
+	import std.conv : to;
+
+	const parts = describeOutput.split("-");
+	const commit = parts[$-1];
+	const num = parts[$-2].to!int;
+	const tag = parts[0 .. $-2].join("-");
+	if (tag.startsWith("v") && isValidVersion(tag[1 .. $])) {
+		if (num == 0) return tag[1 .. $];
+		const i = tag.indexOf('+');
+		auto r = format("%s%scommit.%s.%s", tag[1 .. (i < 0) ? $ : i],
+			parts.length > 3 ? "." : "-", num, commit);
+		if (i > 0) r ~= tag[i .. $];
+		return r;
+	}
+	return null;
+}
+
+unittest {
+	// tag v1.0.0
+	assert(determineVersionFromGitDescribe("v1.0.0-0-deadbeef") == "1.0.0");
+	// 1 commit after v1.0.0
+	assert(determineVersionFromGitDescribe("v1.0.0-1-deadbeef") == "1.0.0-commit.1.deadbeef");
+	// tag v1.0.0+2.0.0
+	assert(determineVersionFromGitDescribe("v1.0.0+2.0.0-0-deadbeef") == "1.0.0+2.0.0");
+	// 12 commits after tag v1.0.0+2.0.0
+	assert(determineVersionFromGitDescribe("v1.0.0+2.0.0-12-deadbeef") == "1.0.0-commit.12.deadbeef+2.0.0");
```

Comment:
What happens with plain `v1.0.0+2.0.0` ?

Reply:
Likewise, `v1.0.0-alpha+alpn`

Reply:
That's no valid `git describe --long --tags` output.

Reply:
If it's run on a tag, you get whatever the tag is, or am I missing something ?

Reply:
Nope, I've tested it - you get `<tag>-0-<hash>` in that case as well. Hence the `0` special case in the original code as well.

Reply:
Perfect, thanks